src/components/SharedComponents/BottomSheet.js

```diff
@@ -1,53 +1,53 @@
 // @flow
 
-import {
-  BottomSheetBackdrop,
-  BottomSheetModal,
-  BottomSheetModalProvider
-} from "@gorhom/bottom-sheet";
+import BottomSheet, { BottomSheetView } from "@gorhom/bottom-sheet";
 import type { Node } from "react";
-import React, { useEffect, useRef } from "react";
+import React, {
+  useCallback, useEffect, useMemo, useRef
+} from "react";
 
 import { viewStyles } from "../../styles/sharedComponents/bottomSheet";
 
 type Props = {
-  children: any
+  children: any,
+  hasScrolled: boolean
 }
 
-const BottomSheet = ( { children }: Props ): Node => {
-  // ref
-  const bottomSheetModalRef = useRef( null );
-
-  const renderBackdrop = props => (
-    <BottomSheetBackdrop
-      // eslint-disable-next-line react/jsx-props-no-spreading
-      {...props}
-      pressBehavior="close"
-      appearsOnIndex={0}
-      disappearsOnIndex={-1}
-    />
-  );
+const StandardBottomSheet = ( { children, hasScrolled }: Props ): Node => {
+  const sheetRef = useRef( null );
+  const snapPoints = useMemo( () => ["45%"], [] );
 
-  useEffect( ( ) => {
-    // opens bottom sheet modal once when component first loads
-    bottomSheetModalRef.current?.present( );
+  // eslint-disable-next-line
+  const noHandle = ( ) => <></>;
+
+  const handleClosePress = useCallback( () => {
+    sheetRef.current?.close();
   }, [] );
 
+  const handleSnapPress = useCallback( ( ) => {
+    sheetRef.current?.snapToIndex( 0 );
+  }, [] );
+
+  useEffect( ( ) => {
+    if ( hasScrolled ) {
+      handleClosePress( );
+    } else {
+      handleSnapPress( );
+    }
+  }, [hasScrolled, handleClosePress, handleSnapPress] );
+
   return (
-    <BottomSheetModalProvider>
-      <BottomSheetModal
-        ref={bottomSheetModalRef}
-        index={0}
-        enableOverDrag={false}
-        enablePanDownToClose={false}
-        snapPoints={["40%"]}
-        backdropComponent={renderBackdrop}
-        style={viewStyles.bottomModal}
-      >
+    <BottomSheet
+      ref={sheetRef}
+      snapPoints={snapPoints}
+      style={viewStyles.shadow}
+      handleComponent={noHandle}
+    >
+      <BottomSheetView style={viewStyles.bottomSheet}>
         {children}
-      </BottomSheetModal>
-    </BottomSheetModalProvider>
+      </BottomSheetView>
+    </BottomSheet>
```

Comment:
You could argue that modal behavior should be optional too, e.g. `<BottomSheet modal />`. You could also make the argument that the behavior is so different that a separate component is justified, e.g. `<ModalBottomSheet />`. Not sure this requires a change in this PR, but something to consider: when you're making a component you know you're going to re-use, try to make the defaults generic but useful, and leave room for enabling expanded functionality. Definitely straddles the line between pre-optimization and prudent planning.

Reply:
I was definitely thinking of modal behavior as belonging in a separate component. It might make sense to create a `<ModalBottomSheet />` component in a separate github issue, since it looks like we have three separate bottom sheet modals on `ObsEdit`, `AddID`, and `ObsDetails`.

Reply:
Let's try to keep that in mind for when we get to those elements.